modules/correlation/radix.c

```diff
@@ -52,23 +52,42 @@ r_parser_string(gchar *str, gint *len, const gchar *param, gpointer state, RPars
 gboolean
 r_parser_qstring(gchar *str, gint *len, const gchar *param, gpointer state, RParserMatch *match)
 {
-  gchar *end;
+  gchar start_char = param[0];
+  gchar stop_char = param[1] ? param[1] : param[0];
+
+  int nesting_level = 0;
+
+  gchar *end = str + 1;
 
-  if ((end = strchr(str + 1, ((gchar *)&state)[0])) != NULL)
+  while (*end)
```

Comment:
while this works I'd try to retain the simplified case where you don't have separate starting/ending quotes.

strchr() is probably _much_ faster than the loop implemented here.

e.g. I'd create two functions, one for the simple case, one for the nested case and decide which one to call in a wrapper that checks param[0] and param[1]